core/src/main/java/org/hibernate/ogm/dialect/spi/GridDialect.java

```diff
@@ -47,16 +47,17 @@
 	 *
 	 * @param key The tuple identifier
 	 * @param tupleContext Contains additional information that might be used to create the tuple
+	 * @param transactionContext Contains information related to the running transaction, it might be null when the db does not require transacitons
 	 * @return the tuple identified by the key
 	 */
-	Tuple getTuple(EntityKey key, TupleContext tupleContext);
+	Tuple getTuple(EntityKey key, TupleContext tupleContext, TransactionContext transactionContext);
```

Comment:
Can you make it part of `TupleContext`? That way `TupleContext` (and similarly, `AssociationContext`) are the entry point into all the information:

* "static" one, as exposed through `AssociationContext#getAssociationTypeContext()`, probably should be done in the same way for the tuple context)
* "runtime" one: operations queue and transaction context